adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -576,32 +594,67 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
     ))
   }
 
-  def loadVcf(filePath: String, sd: Option[SequenceDictionary]): RDD[VariantContext] = {
+  /**
+   * Loads a VCF file into an RDD.
+   *
+   * @param filePath The file to load.
+   * @param sdOpt An optional sequence dictionary, in case the sequence info
+   *              is not included in the VCF.
+   * @return Returns a VariantContextRDD.
+   */
+  def loadVcf(filePath: String,
+              sdOpt: Option[SequenceDictionary] = None): VariantContextRDD = {
+
+    // load vcf data
     val job = HadoopUtil.newJob(sc)
-    val vcc = new VariantContextConverter(sd)
     job.getConfiguration().set("io.compression.codecs", classOf[BGZFCodec].getCanonicalName())
+    val vcc = new VariantContextConverter(sdOpt)
     val records = sc.newAPIHadoopFile(
       filePath,
       classOf[VCFInputFormat], classOf[LongWritable], classOf[VariantContextWritable],
       ContextUtil.getConfiguration(job)
     )
+
+    // attach instrumentation
     if (Metrics.isRecording) records.instrument() else records
 
-    records.flatMap(p => vcc.convert(p._2.get))
+    // load vcf metadata
+    val (vcfSd, samples) = loadVcfMetadata(filePath)
+
+    // we can only replace the sequences header if the sequence info was missing on the vcf
+    require(sdOpt.isEmpty || vcfSd.isEmpty,
+      "Only one of the provided or VCF sequence dictionary can be specified.")
+    val sd = sdOpt.getOrElse(vcfSd)
+
+    VariantContextRDD(records.flatMap(p => vcc.convert(p._2.get)),
+      sd,
+      samples)
   }
 
   def loadParquetGenotypes(
     filePath: String,
     predicate: Option[FilterPredicate] = None,
-    projection: Option[Schema] = None): RDD[Genotype] = {
-    loadParquet[Genotype](filePath, predicate, projection)
+    projection: Option[Schema] = None): GenotypeRDD = {
+    val rdd = loadParquet[Genotype](filePath, predicate, projection)
+
+    // load sequence info
+    val sd = loadAvroSequences(filePath)
+
+    // load avro record group dictionary and convert to samples
+    val rgd = loadAvroSampleMetadata(filePath)
```

Comment:
> I need to study the schema as a whole a bit more before stating an opinion - but if there is even a marginal distinction between SAM/BAM (Record/Read Group) and VCF sample I'd suggest erring on the side of breaking the concepts apart for clarity and adding a new "GenotypedSampleMetadata" avro schema and object to hold VCF sample dict metadata.

I think that sounds reasonable, but maybe we can open that as future work? My general reasoning for starting with a record group is that a VCF sample descends from one or more record groups. I would like to punt this to after this PR for a few reasons:

* This PR resolves a blocking issue for a downstream tool.
* Since we present just a `Seq[String]` with the sample metadata and the actual way that the sample is saved to disk is never visible to the end user, I don't think it is a critical issue.
* We should do a broader refactoring of the various metadata bits --> factor out computational provenance from wetlab provenance, allow VCF sample to descend from read group samples, etc.